src/site/content/en/blog/racer/index.md

```diff
@@ -0,0 +1,289 @@
+    0%     { background-position: 0px 0px; }
+    100%   { background-position: -500px 0px; }
+}
+
+@-webkit-keyframes row2 {
+    0%     { background-position: 0px -100px; }
+    100%   { background-position: -500px -100px; }
+}
+
+@-webkit-keyframes row3 {
+    0%     { background-position: 0px -200px; }
+    100%   { background-position: -500px -200px; }
+}
+```
+
+## Rendering the Cars
+
+As with any car racing game we knew it was important to give the user a feeling of acceleration and handling. Applying a different amount of traction was important for game balancing and the fun factor, so that once a player got a feel for the physics, they’d get a sense of accomplishment and become a better racer. 
+
+Once again we called on Paper.js which comes with an extensive set of math utilities. We used some of its methods to move the car along the path, while adjusting the car position and rotation smoothly each frame.
+
+```js
+var trackOffset = _path.length - (_elapsed % _path.length);
+var trackPoint = _path.getPointAt(trackOffset);
+var trackAngle = _path.getTangentAt(trackOffset).angle;
+
+_velocity.length += _throttle; //apply the throttle
+
+if (!_throttle) {
+    //slow down since the throttle is off
+    _velocity.length *= FRICTION;
+}
+
+if (_velocity.length > MAXVELOCITY) {
+    _velocity.length = MAXVELOCITY;
+}
+
+_velocity.angle = trackAngle;
+
+trackOffset -= _velocity.length;
+_elapsed += _velocity.length;
+
+//find if a lap has been completed
+if (trackOffset < 0) {
+    while (trackOffset < 0) trackOffset += _path.length;
+    trackPoint = _path.getPointAt(trackOffset);
+    console.log('LAP COMPLETE!');
+}
+
+if (_velocity.length > 0.1) {
+    //render the car if there is actually velocity
+    renderCar(trackPoint);
+}
+```
+
+While we were optimizing car rendering, we found an interesting point. On iOS, the best performance was achieved by applying a translate3d transform to the car:
+
+```js
+_car.style.webkitTransform = 'translate3d('+_position.x+'px, '+_position.y+'px, 0px)rotate('+_rotation+'deg)';
+```
+
+On Chrome for Android, the best performance was achieved by calculating the matrix values and applying a matrix transform:
+
+```js
+var rad = _rotation.rotation * (Math.PI * 2 / 360);
+var cos = Math.cos(rad);
+var sin = Math.sin(rad);
+var a = parseFloat(cos).toFixed(8);
+var b = parseFloat(sin).toFixed(8);
+var c = parseFloat(-sin).toFixed(8);
+var d = a;
+_car.style.webkitTransform = 'matrix('+a+', '+b+', '+c+', '+d+', '+_position.x+', '+_position.y+')';
+```
+
+## Keeping the Devices Synced
+
+The most important (and difficult) part of development was to make sure the game synced across devices. We thought users could be forgiving if a car occasionally skipped a few frames because of a slow connection but it wouldn’t be much fun if your car is jumping around, appearing on multiple screens at once.  Resolving this required a ton of trial and error, but we eventually settled on a few tricks which made it work.
```

Comment:
```suggestion
The most important (and difficult) part of development was to make sure the game synced across devices. We thought users could be forgiving if a car occasionally skipped a few frames because of a slow connection but it wouldn't be much fun if your car is jumping around, appearing on multiple screens at once. Resolving this required a ton of trial and error, but we eventually settled on a few tricks which made it work.
```